netket/nn/symmetric_linear.py

```diff
@@ -211,35 +562,219 @@ def full_bias(self, bias):
-            inputs,
+        x = lax.dot_general(
+            x,
             kernel,
-            (((inputs.ndim - 1,), (0,)), ((), ())),
+            (((x.ndim - 1,), (0,)), ((), ())),
             precision=self.precision,
         )
 
         if self.use_bias:
             bias = self.param("bias", self.bias_init, (self.out_features,), self.dtype)
             bias = jnp.asarray(self.full_bias(bias), dtype)
-            y += bias
+            x += bias
+
+        return x
+
+
+def DenseSymm(symmetries, mode="auto", **kwargs):
+    """
+    Implements a projection onto a symmetry group. The output will be
+    equivariant with respect to the symmetry operations in the group and can
+    be averaged to produce an invariant model.
+
+    Args:
+        symmetries: A specification of the symmetry group. Can be given by a
+            nk.graph.Graph, a nk.utils.PermuationGroup, or an array [n_symm, n_sites]
+            specifying the permutations corresponding to symmetry transformations
+            of the lattice.
+        point_group: If symmetries is a graph the default point group of the
+            graph is overwritten from which the space group is build. To only
+            symmetrize over the point_group instead set symmetries=point_group
+        mode: string "fft, matrix, auto" specifying whether to use a fast Fourier
+            transform, matrix multiplication, or to choose a sensible default
+            based on the symmetry group
+        features: The number of symmetry-reduced features. The full output size
+            is [n_symm,features].
+        use_bias: A bool specifying whether to add a bias to the output (default: True).
+        mask: An optional array of shape [n_sites] consisting of ones and zeros
+            that can be used to give the kernel a particular shape
+        dtype: The datatype of the weights. Defaults to a 64bit float
+        precision: Optional argument specifying numerical precision of the computation
+            see `jax.lax.Precision`for details.
+        kernel_init: Optional kernel initialization function. Defaults to variance scaling
+        bias_init: Optional bias initialization function. Defaults to zero initialization
+    """
+
+    if isinstance(symmetries, Graph):
+        kwargs["shape"] = symmetries.extent
+        if "point_group" in kwargs:
+            sym = HashableArray(
+                np.asarray(symmetries.space_group(kwargs["point_group"]))
+            )
+            if mode == "auto":
+                mode = "fft"
+            del kwargs["point_group"]
+        elif symmetries._point_group:
+            sym = HashableArray(np.asarray(symmetries.space_group()))
+            if mode == "auto":
+                mode = "fft"
+        else:
+            if mode == "fft":
+                warnings.warn(
+                    "Graph without a space group specified. Switching to matrix implementation",
+                )
+                mode = "matrix"
```

Comment:
f the user was wrong, hard error.
It's confusing for it to change implementation.

Reply:
fixed